### PR TITLE
TRITON-1995 Add support for encryption to server setup (EDAR)

### PR DESCRIPTION
**Pull request by kusor**
This is strictly to add support for encryption to servers. There will be a later issue for VM's encryption. The screenshots have been added to the JIRA issue and it's working so far with the code awaiting CR for CNAPI here: https://github.com/joyent/sdc-cnapi/pull/19

### CODE REVIEW

**Comment by kusor** [comment]
Obviously, it has no sense to merge this one before we do with the sdc-cnapi one.

File: www/js/tpl/server-setup.hbs [hunk 1]
```diff
@@ -23,6 +23,10 @@
                             <input type="text" class="form-control" name="hostname" />
                             <p class="help-block"><span class="label label-warning">Hostname cannot be changed after setup</span></p>
                         </div>
+
+                        <div class="checkbox">
+                            <label><input class="delegate-dataset" name="encryption_enabled" type="checkbox">Encryption Enabled</label>
```
Comment (twhiteman) [inline www/js/tpl/server-setup.hbs]: Is class="delegate-dataset" the right thing - maybe drop that css class name, or I would think we'd have a more generic CSS class name.

File: www/js/views/server-setup.js [hunk 2]
```diff
@@ -32,7 +32,11 @@ var ServerSetupView = Backbone.Marionette.ItemView.extend({
         var server = this.model;
         var self = this;
         var hostname = this.$('input[name=hostname]').val();
-        this.model.setup({hostname: hostname}, function (job) {
+        var encryption_enabled = this.$('input[name=encryption_enabled]');
+        this.model.setup({
+            hostname: hostname,
+            encryption_enabled: encryption_enabled.is(':checked') || false
```
Comment (twhiteman) [inline www/js/views/server-setup.js]: Pretty sure the 'is()' function will return either true or false, so the "|| false" is not needed.

Comment (kusor) [reply www/js/views/server-setup.js]: Very true.

File: www/js/tpl/server-setup.hbs (hunk 1 above)
Comment (kusor) [reply www/js/tpl/server-setup.hbs]: Yeah, removed.

**Pull request merged**